src/NuGetUtility/ReferencedPackagesReader/ReferencedPackageReader.cs

```diff
@@ -27,9 +29,26 @@ public IEnumerable<PackageIdentity> GetInstalledPackages(string projectPath, boo
                 return Enumerable.Empty<PackageIdentity>();
             }
 
-            return GetInstalledPackagesFromAssetsFile(includeTransitive, project);
+            if (project.IsPackageReferenceProject())
+                return GetInstalledPackagesFromAssetsFile(includeTransitive, project);
+
+            return GetInstalledPackagesFromPackagesConfig(project);
+        }
+
+        private IEnumerable<PackageIdentity> GetInstalledPackagesFromPackagesConfig(IProject project)
```

Comment:
There is a problem with this code: This code does not properly follow transitive dependencies.

Reply:
I was under the impression that special handling of "transitive dependencies" was something that came with PackageReference-format and was very differently handled when using packages.config ; 
With packages.config both direct and transitive dependencies are added at install-time : https://learn.microsoft.com/en-us/nuget/concepts/dependency-resolution#dependency-resolution-with-packagesconfig
Is there something I am missing here?

Reply:
Will see if I can get rid of xml-parsing as suggested

Reply:
OK, right - I'm sorry. I'm not very familiar with packages.config....